api/src/main/java/org/openmrs/api/db/hibernate/HibernateCohortDAO.java

```diff
@@ -51,16 +50,18 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 	public Cohort getCohort(Integer id) throws DAOException {
 		return (Cohort) sessionFactory.getCurrentSession().get(Cohort.class, id);
 	}
-	
+
 	/**
 	 * @see org.openmrs.api.db.CohortDAO#getCohortsContainingPatientId(java.lang.Integer)
 	 */
 	@SuppressWarnings("unchecked")
 	public List<Cohort> getCohortsContainingPatientId(Integer patientId) throws DAOException {
-		Query query = sessionFactory.getCurrentSession().createQuery(
-		    "from Cohort c where :patientId in elements(c.memberIds) and c.voided = false order by name");
-		query.setInteger("patientId", patientId);
-		return (List<Cohort>) query.list();
+		return (List<Cohort>) sessionFactory.getCurrentSession().createCriteria(Cohort.class)
+                .add(Restrictions.eq("voided", false))
```

Comment:
`removePatientFromCohort(cohort, patient)` in CohortService currently calls Cohort.removeMember(). Since this is no longer the case, should there be a new method, say `Cohort.endMembership()` or something?